language-design.md

```diff
@@ -16,6 +16,22 @@ are:
+
+# Quote 3, Steele, 1998
+>"But the key point of the bazaar is not that you can get many persons to work with you at a task, for cathedral builders had a great deal of help, too. Nor is the key point that you get help with the designing as well as with the building, though that in fact is a big win. No, the key point is that in the bazaar style of building a program or designing a language or what you will, the plan can change in real time to meet the needs of those who work on it and use it. This tends to make users stay with it as time goes by; they will take joy in working hard and helping out if they know that their wants and needs have some weight and their hard work can change the plan for the better." [Steele, 1998]
+
+We really liked this quote because we definitely agree that helping with design is sometimes more important than helping with building. We did have a question about it though. Isn't it plausible that if many people are helping with design that the project could lose focus? We think its important for a project to have focus even if it has diversity of ideas. This quote seems to conflict with his later point about not cluttering a language with too many primitives. He discusses the possibility of adding complex/rational numbers, intervals etc. as primitives but states that should definitely not happen. "I might say “yes” to each one of these, but it is clear that I must say “no” to all of them! And so would James Gosling and Bill Joy. To add all these types to the Java programming language would be just too much. Some parts of the programming vocabulary are fit for all programmers to use, but other parts are just for their own niches. It would not be fair to weigh down all programmers with the need to have or to learn all the words for all niche uses." [Steele,1998] We agree with this conclusion and think that people should definitely be able to add to their own "fork of the language" by adding libraries, define first class objects, etc. but we dont think that everyone should be able to design the "master fork" of the language. The analogy Steele uses later of a shopping mall resonated with us because it called both for customizability and diversity in where design ideas come from but also for oversight and accountability on the design/implementation choices made by each devloper.
+
```

Comment:
I agree. In an early iteration, I suspect many people would clutter a language. But if the language is well designed and extensible, it'll be easier to grow it going forward.